diff --git a/Cargo.toml b/Cargo.toml
index 8af9f8e..952b691 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -109,7 +109,7 @@
 wasmtime = { version = "45.0.3", default-features = false }
 x509-parser = "0.18.1"
 yansi = "1.0.1"
-yara-x = { path = "lib", version = "1.20.0" }
+yara-x = { path = "lib", version = "1.20.0", default-features = false }
 yara-x-fmt = { path = "fmt", version = "1.20.0" }
 yara-x-macros = { path = "macros", version = "1.20.0" }
 yara-x-parser = { path = "parser", version = "1.20.0" }
diff --git a/cli/Cargo.toml b/cli/Cargo.toml
index 8e970fb..a0fb961 100644
--- a/cli/Cargo.toml
+++ b/cli/Cargo.toml
@@ -70,7 +70,12 @@
 strum = { workspace = true }
 strum_macros = { workspace = true }
 yansi = { workspace = true }
-yara-x = { workspace = true, features = ["parallel-compilation"] }
+yara-x = { workspace = true, default-features = false, features = [
+    "constant-folding",
+    "exact-atoms",
+    "fast-regexp",
+    "generate-proto-code",
+] }
 yara-x-parser = { workspace = true }
 yara-x-proto = { workspace = true }
 yara-x-fmt = { workspace = true }
diff --git a/Cargo.lock b/Cargo.lock
index 96d38ed..a4a9a85 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -830,8 +830,8 @@ dependencies = [
 [[package]]
 name = "crossbeam-epoch"
-version = "0.9.18"
+version = "0.9.20"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "5b82ac4a3c2ca9c3460964f020e1402edd5753411d7737aa39c3714ad1b5420e"
+checksum = "2d6914041f254d6e9176c01941b21115dcfb7089e55135a35411081bd106ef3f"
 dependencies = [
  "crossbeam-utils",
 ]
@@ -2059,8 +2059,8 @@ dependencies = [
 [[package]]
 name = "memmap2"
-version = "0.9.10"
+version = "0.9.11"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "714098028fe011992e1c3962653c96b2d578c4b4bce9036e15ff220319b1e0e3"
+checksum = "d1219ed1b7f229ee7104d281dd01d6802fe28bb6e95d292942c4daacdeb798c0"
 dependencies = [
  "libc",
 ]
